src/rgw/rgw_coroutine.cc

```diff
@@ -119,6 +120,11 @@ void RGWCompletionManager::_wakeup(void *opaque)
   if (iter != waiters.end()) {
     void *user_id = iter->second;
     waiters.erase(iter);
+    auto it = timer_callbacks.find(opaque);
+    if (it != timer_callbacks.end()) {
+      timer.cancel_event(it->second);
+      timer_callbacks.erase(it);
+    }
```

Comment:
@tianshan I think it's fine. Is there a timer_callbacks lock that we need to care about?:

Reply:
since it's always work with waiters, no need another lock.
but I do find a situation that wakeup by timer in WaitContext::finish() miss the lock, I think change manager->_wakeup to manager->wakeup will fix the problem

Reply:
need to make sure that this actually works and doesn't introduce new lock ordering issue